StudCity.Frontend/src/components/ScrollerPagination/index.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useRef, useEffect, useState } from 'react';
+import Spinner from '../Spinner';
+
+const ScrollerPagination = ({
+	loading,
+	data,
+	ItemComponent,
+	handleClick,
+	handleDispatch,
+	searchData,
+	selectedItems = []
+}) => {
+	const containerRef = useRef(null);
+	const [indexPage, setIndexPage] = useState(1);
+	const [datas, setDatas] = useState([]);
+	const [totalItemsCount, setTotalItemsCount] = useState(0);
+
+	useEffect(() => {
+		if (containerRef.current) {
+			containerRef.current.addEventListener('scroll', handleScroll);
+		}
+		return () => {
+			if (containerRef.current) {
+				containerRef.current.removeEventListener('scroll', handleScroll);
+			}
+		};
+
+	}, [containerRef.current, loading])
+
+	useEffect(() => {
+		if (indexPage > 1 && (totalItemsCount / (10 * (indexPage - 1))) > 1) {
+			handleDispatch(indexPage, searchData);
+		}
+	}, [indexPage])
+
+	useEffect(() => {
+		setDatas([])
+		setIndexPage(1)
+		handleDispatch(1, searchData)
+	}, [searchData])
+
+	useEffect(() => {
+		if (data?.items?.length === 0) {
+			setDatas([])
+		}
+		if (data.items?.length > 0) {
+			setDatas(prev => [...prev, ...data.items])
+			if (indexPage === 1) {
+				setDatas(data.items)
+				setTotalItemsCount(data.pageViewModel.totalItemsCount)
+			}
+		}
+	}, [data.items])
+
+	const handleScroll = () => {
+		if (containerRef.current) {
+			if ((containerRef.current.clientHeight + containerRef.current.scrollTop > containerRef.current.scrollHeight - 1) && !loading) {
```

Comment:
I would suggest before this if statement do some desctructuring of containerRef.current:

> const { clientHeight, scrollTop, scrollHeight } = contianerRef.current;

and then do if without duplicating contianerRef.current, this is purely for readability purposes